test/src/org/omegat/core/segmentation/SRXTest.java

```diff
@@ -197,7 +220,40 @@ public static void testSrxMigration(File segmentConf, File configDir) throws Exc
         assertTrue(srx1.isCascade());
         assertTrue(srx1.isSegmentSubflows());
     }
+    
+   public static class SRXSecurityTest {
+
+        @org.junit.Rule
+        public final LocaleRule localeRule = new LocaleRule(new Locale("en"));
 
+        @org.junit.Rule
+        public final TemporaryFolder folder = TemporaryFolder.builder().assureDeletion().build();
+
+        
+        @Test
+        public void testSRXLoaderSecureCVE_2024_51366() throws IOException {
+            File tmpDir = folder.newFolder();
+            Path segmentConf = tmpDir.toPath().resolve("segmentation.conf");
+            // prepare CVE-2024-51366 exploit code
+            String xmlContent = "<java>\n" +
+                    "    <object class=\"java.lang.ProcessBuilder\">\n" +
```

Comment:
I found it essential that the attack code is not well formatted to prevent pattern matching.

```suggestion
                    "    <object\nclass=\n\"java.lang.ProcessBuilder\">\n" +
```

Reply:
no problem for me